samples/02_power_users_developers/openstreetmap_exploration.ipynb

```diff
@@ -0,0 +1,1338 @@
+{
```

Comment:
<ul><li>In the environment with <em>arcpy</em>,  all cells run successfully, except the last one:</li><li>I continually receive the following error when running the <em>to_featurelayer() </em>command</li></ul><p> </p><p><strong>---------------------------------------------------------------------------</strong></p><p><strong>RuntimeError</strong>                              Traceback (most recent call last)</p><p><strong>&lt;ipython-input-28-e9241c877088&gt;</strong> in &lt;module&gt;</p><p><strong>----&gt; 1 </strong>recycling_hfl <strong>=</strong> recy_df<strong>.</strong>spatial<strong>.</strong>to_featurelayer<strong>(f'OSM_Recycling_{round(time.time())}',</strong> gis<strong>=</strong>gis<strong>,</strong> tags<strong>='OSM')</strong></p><p>      2 shops_hfl <strong>=</strong> shop_df<strong>.</strong>spatial<strong>.</strong>to_featurelayer<strong>(f'OSM_Shops_{round(time.time())}',</strong> gis<strong>=</strong>gis<strong>,</strong> tags<strong>='OSM')</strong></p><p>      3 </p><p>      4 display<strong>(</strong>recycling_hfl<strong>)</strong></p><p>      5 display<strong>(</strong>shops_hfl<strong>)</strong></p><p> </p><p><strong>~\AppData\Local\ESRI\conda\envs\arcgispro-py3-clone\lib\site-packages\arcgis\features\geo\_accessor.py</strong> in to_featurelayer<strong>(self, title, gis, tags, folder)</strong></p><p>   1983                 <strong>raise</strong> ValueError<strong>("GIS object must be provided")</strong></p><p>   1984         content <strong>=</strong> gis<strong>.</strong>content</p><p><strong>-&gt; 1985         return</strong> content<strong>.</strong>import_data<strong>(</strong>self<strong>.</strong>_data<strong>,</strong> folder<strong>=</strong>folder<strong>,</strong> title<strong>=</strong>title<strong>,</strong> tags<strong>=</strong>tags<strong>)</strong></p><p>   1986     <strong># ----------------------------------------------------------------------</strong></p><p>   1987     <strong>@</strong>staticmethod</p><p> </p><p><strong>~\AppData\Local\ESRI\conda\envs\arcgispro-py3-clone\lib\site-packages\arcgis\gis\__init__.py</strong> in import_data<strong>(self, df, address_fields, folder, **kwargs)</strong></p><p>   4114                     ds <strong>=</strong>\</p><p>   4115                         df.spatial.to_featureclass(location=os.path.join(fgdb,</p><p><strong>-&gt; 4116                                                                          os.path.basename(temp_dir)))</strong></p><p>   4117 </p><p>   4118                 zip_fgdb <strong>=</strong> zipws<strong>(</strong>path<strong>=</strong>fgdb<strong>,</strong> outfile<strong>=</strong>temp_zip<strong>,</strong> keep<strong>=True)</strong></p><p> </p><p><strong>~\AppData\Local\ESRI\conda\envs\arcgispro-py3-clone\lib\site-packages\arcgis\features\geo\_accessor.py</strong> in to_featureclass<strong>(self, location, overwrite)</strong></p><p>   1923         return to_featureclass(self,</p><p>   1924                                location<strong>=</strong>location<strong>,</strong></p><p><strong>-&gt; 1925                                overwrite=overwrite)</strong></p><p>   1926     <strong>#----------------------------------------------------------------------</strong></p><p>   1927     <strong>def</strong> to_table<strong>(</strong>self<strong>,</strong> location<strong>,</strong> overwrite<strong>=True):</strong></p><p> </p><p><strong>~\AppData\Local\ESRI\conda\envs\arcgispro-py3-clone\lib\site-packages\arcgis\features\geo\_io\fileops.py</strong> in to_featureclass<strong>(geo, location, overwrite, validate)</strong></p><p>    610 </p><p>    611         array <strong>=</strong> np<strong>.</strong>array<strong>([],</strong> np<strong>.</strong>dtype<strong>(</strong>dtypes<strong>))</strong></p><p><strong>--&gt; 612         </strong>arcpy<strong>.</strong>da<strong>.</strong>ExtendTable<strong>(</strong>fc<strong>,</strong> oidfld<strong>,</strong> array<strong>,</strong> join_dummy<strong>,</strong> append_only<strong>=False)</strong></p><p>    613 </p><p>    614         <strong># 3. Insert the Data</strong></p><p> </p><p><strong>RuntimeError</strong>: The operation was attempted on an empty geometry.</p><p> </p><ul><li>I tried altering the geometry column by running <em>df.spatial.set_geometry(col='SHAPE')</em>, but I still received the error</li><li>I also received an error (but have been unable to reproduce) about a Field name being invalid, which I assumed resulted because of the colons in certain field names in the <em>shop_df</em> and <em>recy_df</em> frames. I replaced the colons with underscores and still received the above conda error.</li></ul><p> </p><ul><li>in the environment without arcpy, the feature layers are created correctly</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/551/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>